### ISSUE TITLE
Use of str_interp

### ISSUE THREAD

**Issue by jonocarroll** [hadley/r4ds#206]
Section 11 on Strings has the following example 

```
match <- str_interp("(${num}) ([^ ]+s)\\b")
sentences %>% 
  str_subset(match) %>% 
  head(10) %>% 
  str_match(match)
```

`str_interp` is not mentioned anywhere else in the book and I was initially tripped up by the fact that it's not included in the CRAN 1.0.0 release of `stringr`. While I appreciate that it's not too difficult to realise that the dev version of `stringr` is required in order to look up what exactly that function does, it's a highly non-trivial complication that is critical to understanding the example.

I'd like to suggest either changing the example or adding a small section on what `str_interp` does.

**Comment by hadley** [comment]
Hmmmm, I don't even understand the point of that example - there's no text explaining it and it seems a bit overwrought for just explaining `str_match()`.

**Comment by jonocarroll** [comment]
Commit 0ab8d322fb004b8b8b7d5f833602aa518b9309dc closes #206.
